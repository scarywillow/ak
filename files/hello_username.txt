  section_data
  question . What is your name?
  hello . Hello,
  mark . !
  section_program
  start:
    LD question R6
    LD hello R7
    LD mark R8
    out R6
    IN
    OUT R7
    OUT R4
    OUT R8
    HLT
