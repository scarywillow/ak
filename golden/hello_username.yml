in_source: |
  section_data
  question . What is your name?
  hello . Hello,
  mark . !
  section_program
  start:
    LD question R6
    LD hello R7
    LD mark R8
    OUT R6
    IN
    OUT R7
    OUT R4
    OUT R8
    HLT
in_stdin: '"A","l","i","c","e"'
out_code: |
  Instruction counter: 0, LD ['question', 'R6']
  Instruction counter: 1, LD ['hello', 'R7']
  Instruction counter: 2, LD ['mark', 'R8']
  Instruction counter: 3, OUT ['R6']
  Instruction counter: 4, IN []
  Instruction counter: 5, OUT ['R7']
  Instruction counter: 6, OUT ['R4']
  Instruction counter: 7, OUT ['R8']
  Instruction counter: 8, HLT []
out_stdout: 'What is your name?Hello,Alice!'
out_log: |
  DEBUG    root:machine.py:250 LD ['question', 'R6'] {PC: 0, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: , REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:250 LD ['hello', 'R7'] {PC: 1, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: , REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:250 LD ['mark', 'R8'] {PC: 2, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: , REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:250 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: []}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e']}
  DEBUG    root:machine.py:198 OUT ['R6'] {PC: 3, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:250 IN [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: , REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:205 IN [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: A, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:205 IN [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: Al, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:205 IN [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: Ali, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:205 IN [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: Alic, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:205 IN [] {PC: 4, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:250 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?']}
  DEBUG    root:machine.py:198 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H']}
  DEBUG    root:machine.py:198 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e']}
  DEBUG    root:machine.py:198 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l']}
  DEBUG    root:machine.py:198 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l']}
  DEBUG    root:machine.py:198 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o']}
  DEBUG    root:machine.py:198 OUT ['R7'] {PC: 5, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',']}
  DEBUG    root:machine.py:250 OUT ['R4'] {PC: 6, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',']}
  DEBUG    root:machine.py:198 OUT ['R4'] {PC: 6, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A']}
  DEBUG    root:machine.py:198 OUT ['R4'] {PC: 6, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l']}
  DEBUG    root:machine.py:198 OUT ['R4'] {PC: 6, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l', 'i']}
  DEBUG    root:machine.py:198 OUT ['R4'] {PC: 6, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l', 'i', 'c']}
  DEBUG    root:machine.py:198 OUT ['R4'] {PC: 6, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l', 'i', 'c', 'e']}
  DEBUG    root:machine.py:250 OUT ['R8'] {PC: 7, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l', 'i', 'c', 'e']}
  DEBUG    root:machine.py:198 OUT ['R8'] {PC: 7, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l', 'i', 'c', 'e', '!']}
  DEBUG    root:machine.py:250 HLT [] {PC: 8, REG0: 0, REG1: , REG2: , REG3: , REG4: Alice, REG5: , REG6: What is your name?, REG7: Hello,, REG8: !, REG9: , REG10: , REG11: , REG12: , REG13: , REG14: , REG15: , OUTPUT_PORT: ['W', 'h', 'a', 't', ' ', 'i', 's', ' ', 'y', 'o', 'u', 'r', ' ', 'n', 'a', 'm', 'e', '?', 'H', 'e', 'l', 'l', 'o', ',', 'A', 'l', 'i', 'c', 'e', '!']}
